From 16603c6af96bd3daef011df8c7e42b7335c07ef3 Mon Sep 17 00:00:00 2001
From: Michael Bestas <mikeioannina@gmail.com>
Date: Tue, 30 Apr 2013 02:35:46 +0300
Subject: [PATCH] mogami: remove old rfkill implementation

BT is managed by BT_WILINK & TI_ST drivers now

Signed-off-by: Michael Bestas <mikeioannina@gmail.com>

Change-Id: I629c8b098ee85a8b4248d7d3eacada43360dd5f0
---
 arch/arm/mach-msm/Makefile                   |  22 ++---
 arch/arm/mach-msm/board-semc_mogami-rfkill.c | 117 ---------------------------
 2 files changed, 11 insertions(+), 128 deletions(-)
 delete mode 100644 arch/arm/mach-msm/board-semc_mogami-rfkill.c

diff --git a/arch/arm/mach-msm/Makefile b/arch/arm/mach-msm/Makefile
index c42033a..cbaf2e9 100644
--- a/arch/arm/mach-msm/Makefile
+++ b/arch/arm/mach-msm/Makefile
@@ -248,17 +248,17 @@ obj-$(CONFIG_ARCH_MSM9615) += board-9615.o devices-9615.o board-9615-regulator.o
 obj-$(CONFIG_ARCH_MSM9615) += clock-local.o clock-9615.o acpuclock-9615.o clock-rpm.o
 obj-$(CONFIG_ARCH_MSMCOPPER) += board-copper.o board-dt.o
 
-obj-$(CONFIG_MACH_SEMC_MOGAMI) += board-semc_mogami.o board-semc_mogami-rfkill.o board-semc_mogami-wifi.o
-
-obj-$(CONFIG_MACH_SEMC_SMULTRON) += board-semc_mogami.o leds-smultron.o vibrator-mogami.o charger-smultron.o board-semc_mogami-wifi.o touch-smultron.o gpio-smultron.o keypad-smultron.o board-semc_mogami-rfkill.o
-obj-$(CONFIG_MACH_SEMC_COCONUT) += board-semc_mogami.o leds-coconut.o vibrator-mogami.o charger-coconut.o board-semc_mogami-wifi.o touch-coconut.o gpio-coconut.o keypad-coconut.o board-semc_mogami-rfkill.o
-obj-$(CONFIG_MACH_SEMC_HAIDA) += board-semc_mogami.o leds-haida.o vibrator-mogami.o charger-haida.o board-semc_mogami-rfkill.o keypad-haida.o gpio-haida.o board-semc_mogami-wifi.o
-obj-$(CONFIG_MACH_SEMC_URUSHI) += board-semc_mogami.o leds-urushi.o vibrator-mogami.o charger-urushi.o board-semc_mogami-wifi.o gpio-urushi.o keypad-urushi.o board-semc_mogami-rfkill.o
-obj-$(CONFIG_MACH_SEMC_MANGO) += board-semc_mogami.o leds-mango.o vibrator-mogami.o charger-mango.o board-semc_mogami-wifi.o touch-mango.o gpio-mango.o keypad-mango.o board-semc_mogami-rfkill.o
-obj-$(CONFIG_MACH_SEMC_ANZU) += board-semc_mogami.o board-semc_mogami-rfkill.o keypad-anzu.o leds-anzu.o  vibrator-mogami.o gpio-anzu.o charger-anzu.o board-semc_mogami-wifi.o
-obj-$(CONFIG_MACH_SEMC_IYOKAN) += board-semc_mogami.o leds-iyokan.o vibrator-mogami.o charger-iyokan.o board-semc_mogami-wifi.o touch-iyokan.o gpio-iyokan.o keypad-iyokan.o board-semc_mogami-rfkill.o
-obj-$(CONFIG_MACH_SEMC_HALLON) += board-semc_mogami.o leds-hallon.o vibrator-mogami.o charger-hallon.o board-semc_mogami-rfkill.o keypad-hallon.o gpio-hallon.o board-semc_mogami-wifi.o
-obj-$(CONFIG_MACH_SEMC_SATSUMA) += board-semc_mogami.o board-semc_mogami-rfkill.o keypad-satsuma.o leds-satsuma.o gpio-satsuma.o charger-satsuma.o board-semc_mogami-wifi.o touch-satsuma.o vibrator-mogami.o
+obj-$(CONFIG_MACH_SEMC_MOGAMI) += board-semc_mogami.o board-semc_mogami-wifi.o
+
+obj-$(CONFIG_MACH_SEMC_SMULTRON) += board-semc_mogami.o board-semc_mogami-wifi.o vibrator-mogami.o charger-smultron.o gpio-smultron.o keypad-smultron.o leds-smultron.o touch-smultron.o
+obj-$(CONFIG_MACH_SEMC_COCONUT) += board-semc_mogami.o board-semc_mogami-wifi.o vibrator-mogami.o charger-coconut.o gpio-coconut.o keypad-coconut.o leds-coconut.o touch-coconut.o
+obj-$(CONFIG_MACH_SEMC_HAIDA) += board-semc_mogami.o board-semc_mogami-wifi.o vibrator-mogami.o charger-haida.o gpio-haida.o keypad-haida.o leds-haida.o
+obj-$(CONFIG_MACH_SEMC_URUSHI) += board-semc_mogami.o board-semc_mogami-wifi.o vibrator-mogami.o charger-urushi.o gpio-urushi.o keypad-urushi.o leds-urushi.o
+obj-$(CONFIG_MACH_SEMC_MANGO) += board-semc_mogami.o board-semc_mogami-wifi.o vibrator-mogami.o charger-mango.o gpio-mango.o keypad-mango.o leds-mango.o touch-mango.o
+obj-$(CONFIG_MACH_SEMC_ANZU) += board-semc_mogami.o board-semc_mogami-wifi.o vibrator-mogami.o charger-anzu.o gpio-anzu.o keypad-anzu.o leds-anzu.o
+obj-$(CONFIG_MACH_SEMC_IYOKAN) += board-semc_mogami.o board-semc_mogami-wifi.o vibrator-mogami.o charger-iyokan.o gpio-iyokan.o keypad-iyokan.o leds-iyokan.o touch-iyokan.o
+obj-$(CONFIG_MACH_SEMC_HALLON) += board-semc_mogami.o board-semc_mogami-wifi.o vibrator-mogami.o charger-hallon.o gpio-hallon.o keypad-hallon.o leds-hallon.o
+obj-$(CONFIG_MACH_SEMC_SATSUMA) += board-semc_mogami.o board-semc_mogami-wifi.o vibrator-mogami.o charger-satsuma.o gpio-satsuma.o keypad-satsuma.o leds-satsuma.o touch-satsuma.o
 
 obj-$(CONFIG_MACH_SAPPHIRE) += board-sapphire.o board-sapphire-gpio.o
 obj-$(CONFIG_MACH_SAPPHIRE) += board-sapphire-keypad.o board-sapphire-panel.o
diff --git a/arch/arm/mach-msm/board-semc_mogami-rfkill.c b/arch/arm/mach-msm/board-semc_mogami-rfkill.c
deleted file mode 100644
index dfc2261..0000000
--- a/arch/arm/mach-msm/board-semc_mogami-rfkill.c
+++ /dev/null
@@ -1,117 +0,0 @@
-/*
- * Copyright (C) 2009 Google, Inc.
- * Copyright (C) 2009 HTC Corporation.
- * Copyright (C) 2010 Sony Ericsson Mobile Communications AB
- *
- * This software is licensed under the terms of the GNU General Public
- * License version 2, as published by the Free Software Foundation, and
- * may be copied, distributed, and modified under those terms.
- *
- * This program is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- * GNU General Public License for more details.
- *
- */
-
-#include <linux/delay.h>
-#include <linux/device.h>
-#include <linux/module.h>
-#include <linux/platform_device.h>
-#include <linux/rfkill.h>
-#include <asm/gpio.h>
-#include <asm/mach-types.h>
-
-static struct rfkill *bt_rfk;
-static const char bt_name[] = "wl127x";
-
-struct rfkill_gpio_data {
-	int gpio;
-};
-
-static int bluetooth_set_power(void *data, bool blocked)
-{
-	int ret = -EINVAL;
-	int (*power_control)(int enable);
-
-	if (data) {
-		power_control = data;
-		ret = (power_control)(!blocked);
-	}
-	return ret;
-}
-
-static struct rfkill_ops mogami_rfkill_ops = {
-	.set_block = bluetooth_set_power,
-};
-
-static int mogami_rfkill_probe(struct platform_device *pdev)
-{
-	int rc = 0;
-	bool default_state = true;  /* off */
-
-	rc = bluetooth_set_power(pdev->dev.platform_data, default_state);
-	if (rc)
-		goto err_rfkill;
-
-	bt_rfk = rfkill_alloc(bt_name, &pdev->dev, RFKILL_TYPE_BLUETOOTH,
-				&mogami_rfkill_ops, pdev->dev.platform_data);
-	if (!bt_rfk) {
-		rc = -ENOMEM;
-		goto err_rfkill;
-	}
-
-	rfkill_set_states(bt_rfk, default_state, false);
-
-	/* userspace cannot take exclusive control */
-
-	rc = rfkill_register(bt_rfk);
-	if (rc)
-		goto err_rfkill_reg;
-
-	platform_set_drvdata(pdev, bt_rfk);
-	return 0;
-
-err_rfkill_reg:
-	rfkill_destroy(bt_rfk);
-err_rfkill:
-	return rc;
-}
-
-static int mogami_rfkill_remove(struct platform_device *pdev)
-{
-	struct rfkill_gpio_data *gdata = platform_get_drvdata(pdev);
-
-	rfkill_unregister(bt_rfk);
-	rfkill_destroy(bt_rfk);
-	gpio_free(gdata->gpio);
-	kfree(gdata);
-	platform_set_drvdata(pdev, NULL);
-
-	return 0;
-}
-
-static struct platform_driver mogami_rfkill_driver = {
-	.probe = mogami_rfkill_probe,
-	.remove = mogami_rfkill_remove,
-	.driver = {
-		.name = "mogami-rfkill",
-		.owner = THIS_MODULE,
-	},
-};
-
-static int __init mogami_rfkill_init(void)
-{
-	return platform_driver_register(&mogami_rfkill_driver);
-}
-
-static void __exit mogami_rfkill_exit(void)
-{
-	platform_driver_unregister(&mogami_rfkill_driver);
-}
-
-module_init(mogami_rfkill_init);
-module_exit(mogami_rfkill_exit);
-MODULE_DESCRIPTION("mogami rfkill");
-MODULE_AUTHOR("Vitaly Wool <vitaly.wool@sonyericsson.com>");
-MODULE_LICENSE("GPL");
-- 
1.8.1.6

